src/object/cloud_particle_system.cpp

```diff
@@ -135,15 +199,31 @@ void CloudParticleSystem::update(float dt_sec)
   // Clear dead clouds.
   // Iterate through the vector backwards to avoid affecting the index of elements
   // after removal, preventing buggy behavior.
-  for (int i = static_cast<int>(particles.size()) - 1; i >= 0; --i) {
+  for (int i = static_cast<int>(particles.size()) - 1; i >= 0; --i)
+  {
     auto particle = dynamic_cast<CloudParticle*>(particles.at(i).get());
-    
+
     if (particle->target_alpha == 0.f && particle->target_time_remaining == 0.f)
-      particles.erase(particles.begin()+i);
+      particles.erase(particles.begin() + i);
   }
 }
 
-int CloudParticleSystem::add_clouds(int amount, float fade_time)
+void
+CloudParticleSystem::apply_fog_effect(DrawingContext& context)
+{
+  float opacity = fog_max_value * (static_cast<float>(m_current_amount) - static_cast<float>(fog_start_amount)) / (static_cast<float>(max_amount) - static_cast<float>(fog_start_amount));
```

Comment:
What's going on with all the casting? These are floats. Is it because they're const? Constants should go to the top of this cpp file and should be marked constexpr instead.

Reply:
Iirc I had some sort of warning that made VS refuse to compile